src/main/scala/org/constellation/primitives/CheckpointBlock.scala

```diff
@@ -7,7 +7,62 @@ import cats.implicits._
 import constellation.signedObservationEdge
 import org.constellation.DAO
 import org.constellation.primitives.Schema._
-import org.constellation.util.HashSignature
+import org.constellation.primitives.storage.StorageService
+import org.constellation.util.{HashSignature, MerkleTree}
+
+object CheckpointBlockData {
+
+  def apply(data: CheckpointBlock)(implicit dao: DAO): CheckpointBlockData = {
+    CheckpointBlockData(
+      add(data.transactions.map(_.hash), dao.transactionService.merklePool).get,
```

Comment:
You shouldn't use statements which has side effects inside, for creating a case class